pkgs/development/libraries/xed/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, python3, python3Packages }:
+
+let
+  # mbuild is a custom build system used only to build xed
+  mbuild = python3Packages.buildPythonPackage rec {
+    pname = "mbuild";
+    version = "0.2496-dev";
+
+    src = fetchFromGitHub {
+      owner = "intelxed";
+      repo = "mbuild";
+      rev = "3e8eb33aada4153c21c4261b35e5f51f6e2019e8";
+      sha256 = "0yamgzkzw4v6x1a857psw9f7i62ydgd0zaqrf33dbdg8hfd2mq3q";
+    };
+  };
+
+  pythonMbuild = python3.withPackages (_: [ mbuild ]);
+
+in stdenv.mkDerivation rec {
+  pname = "xed";
+  version = "12.0.1";
+
+  src = fetchFromGitHub {
+    owner = "intelxed";
+    repo = "xed";
+    rev = version;
+    sha256 = "07zfff8zf29c2n0wal87hiqfq3cwcjn80zz78mz0nyjfj09nd39f";
+  };
+
+  buildPhase = ''
+    ${pythonMbuild}/bin/python ./mfile.py test --prefix $out
+  '';
```

this might work, the interpreter gets propagated for `buildPythonPackage`
```suggestion
in stdenv.mkDerivation rec {
  pname = "xed";
  version = "12.0.1";

  src = fetchFromGitHub {
    owner = "intelxed";
    repo = "xed";
    rev = version;
    sha256 = "07zfff8zf29c2n0wal87hiqfq3cwcjn80zz78mz0nyjfj09nd39f";
  };

  nativeBuildInputs = [ mbuild ];
```